File: templates/problems/01_arrays_hashing/09_longest_consecutive/python/solution.py
def solve_longest_consecutive(nums: list[int]) -> int:
    """Find the longest consecutive sequence using a hash set.

    Time: O(n), Space: O(n)
    """
    num_set = set(nums)

    best = 0
    for n in num_set:
        # Only start counting from the beginning of a sequence
        if n - 1 in num_set:
            continue
        length = 1
        while n + length in num_set:
            length += 1
        if length > best:
            best = length
    return best

File: templates/problems/01_arrays_hashing/09_longest_consecutive/typescript/solution.ts
/**
 * Find the longest consecutive sequence using a hash set.
 * Time: O(n), Space: O(n)
 */
export function solveLongestConsecutive(nums: number[]): number {
    const numSet = new Set<number>(nums);

    let best = 0;
    for (const n of numSet) {
        // Only start counting from the beginning of a sequence
        if (numSet.has(n - 1)) {
            continue;
        }
        let length = 1;
        while (numSet.has(n + length)) {
            length++;
        }
        if (length > best) {
            best = length;
        }
    }
    return best;
}
